apps/glusterfs/dbcommon_test.go

```diff
@@ -0,0 +1,216 @@
+//
+
+package glusterfs
+
+import (
+	"os"
+	"testing"
+
+	"github.com/boltdb/bolt"
+	"github.com/heketi/heketi/pkg/glusterfs/api"
+	"github.com/heketi/heketi/pkg/utils"
+	"github.com/heketi/tests"
+)
+
+func TestDeleteBricksWithEmptyPath(t *testing.T) {
+	tmpfile := tests.Tempfile()
+	defer os.Remove(tmpfile)
+
+	app := NewTestApp(tmpfile)
+	defer app.Close()
+
+	err := setupSampleDbWithTopology(app,
+		1,    // clusters
+		3,    // nodes_per_cluster
+		1,    // devices_per_node,
+		5*TB, // disksize
+	)
+	tests.Assert(t, err == nil, "expected err == nil, got:", err)
+
+	var nodeEntry *NodeEntry
+	vreq := &api.VolumeCreateRequest{}
+	vreq.Size = 100
+	vreq.Durability.Type = api.DurabilityReplicate
+	vreq.Durability.Replicate.Replica = 3
+	// create a few volumes
+	for i := 0; i < 15; i++ {
+		v := NewVolumeEntryFromRequest(vreq)
+		err = v.Create(app.db, app.executor, app.Allocator())
+		tests.Assert(t, err == nil, "expected err == nil, got:", err)
+	}
+
+	// grab a device that has bricks
+	var d *DeviceEntry
+	var newbrick *BrickEntry
+	err = app.db.View(func(tx *bolt.Tx) error {
+		dl, err := DeviceList(tx)
+		if err != nil {
+			return err
+		}
+		for _, id := range dl {
+			d, err = NewDeviceEntryFromId(tx, id)
+			if err != nil {
+				return err
+			}
+			if len(d.Bricks) > 0 {
+				return nil
+			}
+		}
+		t.Fatalf("should have at least one device with bricks")
+		return nil
+	})
+
+	// At this point, we have 15 legal bricks on each device and we have
+	// made note of a device(node and cluster in corollary) where to will
+	// create 25 bricks with empty path.
+
+	// CASE1: use "all" bool to delete empty path bricks from all clusters
+	// ====================================================================
+	// create bricks in device and set the path empty
+	// save device and brick to db
+	for i := 0; i < 25; i++ {
+		newbrick = d.NewBrickEntry(102400, 1, 2000, utils.GenUUID())
+		newbrick.Info.Path = ""
+		d.BrickAdd(newbrick.Id())
+		err = app.db.Update(func(tx *bolt.Tx) error {
+			err = d.Save(tx)
+			tests.Assert(t, err == nil)
+			return newbrick.Save(tx)
+		})
+		tests.Assert(t, err == nil)
+	}
+	err = app.db.View(func(tx *bolt.Tx) error {
+		d, err = NewDeviceEntryFromId(tx, d.Info.Id)
+		return err
+	})
+	tests.Assert(t, err == nil, "expected err == nil, got:", err)
+	tests.Assert(t, len(d.Bricks) == 40,
+		"expected len(d.Bricks) == 40, got:", len(d.Bricks))
+
+	// now delete bricks with empty path
+	err = DeleteBricksWithEmptyPath(app.db, true, []string{}, []string{}, []string{}, true)
+	tests.Assert(t, err == nil, "expected err == nil, got:", err)
+
+	err = app.db.View(func(tx *bolt.Tx) error {
+		d, err = NewDeviceEntryFromId(tx, d.Info.Id)
+		return err
+	})
+	tests.Assert(t, err == nil, "expected err == nil, got:", err)
+	tests.Assert(t, len(d.Bricks) == 15,
+		"expected len(d.Bricks) == 15, got:", len(d.Bricks))
+
+	// CASE2: delete empty path bricks from this device
```

Comment:
This feels like it should be a 2nd test case function. You could move the setup stuff into a common function like "setupClusterForEmptyPathBricks" or something.